src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -1452,4 +1602,199 @@ RppStatus rppt_vignette_gpu(RppPtr_t srcPtr,
 #endif // backend
 }
 
+/******************** pixelate ********************/
+
+RppStatus rppt_pixelate_gpu(RppPtr_t srcPtr,
+                            RpptDescPtr srcDescPtr,
+                            RppPtr_t dstPtr,
+                            RpptDescPtr dstDescPtr,
+                            Rpp32f pixelationPercentage,
+                            RpptROIPtr roiTensorPtrSrc,
+                            RpptRoiType roiType,
+                            rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+
+    RpptInterpolationType interpolationType = RpptInterpolationType::BILINEAR;
+    RpptDesc interDesc;
+    interDesc = *srcDescPtr;
+    RpptDescPtr interDescPtr = &interDesc;
+    RpptImagePatchPtr internalDstImgSizes;
+    hipHostMalloc(&internalDstImgSizes, sizeof(RpptImagePatch) * rpp::deref(rppHandle).GetBatchSize());
+    RpptROIPtr internalRoiTensorPtrSrc;
+    hipHostMalloc(&internalRoiTensorPtrSrc, sizeof(RpptROI) * rpp::deref(rppHandle).GetBatchSize());
+
+    for (int i = 0; i < srcDescPtr->n; i++)
+    {
+        internalRoiTensorPtrSrc[i].xywhROI.roiWidth = roiTensorPtrSrc[i].xywhROI.roiWidth;
+        internalRoiTensorPtrSrc[i].xywhROI.roiHeight = roiTensorPtrSrc[i].xywhROI.roiHeight;
+        internalDstImgSizes[i].width = (roiTensorPtrSrc[i].xywhROI.roiWidth * (100 - pixelationPercentage)) / 100;
+        internalDstImgSizes[i].height = (roiTensorPtrSrc[i].xywhROI.roiHeight * (100 - pixelationPercentage)) / 100;
+    }
+
+    // Optionally set w stride as a multiple of 8 for src/dst
+    interDescPtr->h = (interDescPtr->h * (100 - pixelationPercentage)) / 100;
+    interDescPtr->w = (((interDescPtr->w * (100 - pixelationPercentage)) / 100 * 8) * 8) + 8;
+    interDescPtr->strides.nStride = interDescPtr->w * interDescPtr->h * interDescPtr->c;
+    // The stride changes with the change in the height and width
+    if(srcDescPtr->layout == RpptLayout::NCHW)
+    {
+        interDescPtr->strides.cStride = interDescPtr->w * interDescPtr->h;
+        interDescPtr->strides.hStride = interDescPtr->w;
+    }
+    if (srcDescPtr->layout == RpptLayout::NHWC)
+    {
+        interDescPtr->strides.cStride = 1;
+        interDescPtr->strides.hStride = interDescPtr->c * interDescPtr->w;
+    }
+    RppPtr_t interDstPtr;
+    Rpp64u interBufferSize = static_cast<Rpp64u>(interDescPtr->h) * static_cast<Rpp64u>(interDescPtr->w) * static_cast<Rpp64u>(srcDescPtr->c) * static_cast<Rpp64u>(srcDescPtr->n);
```

Comment:
We cannot have lines L1635-L1651 copied similar to the test suite here! Pls re-evaluate or find another way.